File: 2419.java
class Solution {
    public int longestSubarray(int[] nums) {
        int max = 0, result = 0, current = 0;
        for (int num : nums) {
            if (num > max) {
                max = num;
                result = 0;
                current = 0;
            }

            if (max == num) {
                current++;
            } else {
                current = 0;
            }

            result = Math.max(result, current);
        }
        return result;
    }
}

File: Python/2419.py
from typing import List


class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        max_val = result = consecutive_count = 0
        for num in nums:
            if max_val < num:
                max_val = num
                result = consecutive_count = 0

            if max_val == num:
                consecutive_count += 1
            else:
                consecutive_count = 0

            result = max(result, consecutive_count)

        return result
